feat(namespacing): Namespace Native Octagon Components to avoid collisions

andrewleyva opened the pull request:
# problem statement

We could have style collisions if we don't properly namespace.

# solution

namespace Native Octagon components with .octagon.

ggascoigne commented on src/styles/components/large-card.css:
```diff
@@ -3,7 +3,7 @@
   font-size: 20px !important;
 }
-.large_card {
+.octagon.large_card {
```
Since this is in cssnext, wouldn't it make more sense to put both of these inside a single .octagon section?

andrewleyva replied:
👍

andrewleyva commented:
Yes, more are going to be added. This isn't ready for merge or review yet. 
I'm not sure why I'm getting a snap diff when nothing has changed visually so i wanted to push it up to GH and see if the same was occurring. It is, which is odd.

wa11-e commented:
Woohoo! new surge deployment available for viewing! :tada: [surgereviewbaojfwbswnrhezo.surge.sh](http://surgereviewbaojfwbswnrhezo.surge.sh)

wa11-e commented:
Woohoo! new surge deployment available for viewing! :tada: [surgerevieweehuxibyqongukh.surge.sh](http://surgerevieweehuxibyqongukh.surge.sh)

cdaringe commented:
i'm not saying it's worth it, but i _wonder_ if it's worth maybe scripting the addition of the className to all root components, _maybe_?  i dont like that there's magic (e.g. a hidden class not shown in code), but it's worth thinking about.

andrewleyva commented:
@cdaringe It's an interesting idea, and would make it easier to change the name if for some reason we needed to in the future.
To do this though, we'd need to restructure quite a bit of the CSS and some of the JSX. There is at least one case where the namespace is added in a different manner than the other components.
Tag Button:
  let tagClass = 'octagon tagContainer'
  if (props.selectedTagId === props.tag.id) {
    tagClass = 'selectedTagContainer'
  }

wa11-e commented:
Woohoo! new surge deployment available for viewing! :tada: [surgereviewbsjybzqzseslalx.surge.sh](http://surgereviewbsjybzqzseslalx.surge.sh)

wa11-e commented:
Woohoo! new surge deployment available for viewing! :tada: [surgereviewwqdrcjtxjsybqlc.surge.sh](http://surgereviewwqdrcjtxjsybqlc.surge.sh)

wa11-e commented:
Woohoo! new surge deployment available for viewing! :tada: [surgerevieweriorafinktjlyh.surge.sh](http://surgerevieweriorafinktjlyh.surge.sh)

andrewleyva commented:
@ggascoigne @cdaringe all native components with styles that could collide have been namespaced w/ 'octagon'.

cdaringe commented on src/components/native/StopStartButton/StopStartButton.jsx:
```diff
@@ -4,11 +4,11 @@ import Icon from '../../suir/icon/Icon'
+    iconClass += ' ei icon_pause_alt2 stop-start-button'
   } else {
-    iconClass = 'ei arrow_triangle-right_alt2 stop-start-button'
+    iconClass += ' ei arrow_triangle-right_alt2 stop-start-button'
```
no change required or requested, but rather than have to worry about whitespace prefixing and postfixing, using templates removes the issue entirely:

```js
let iconClass = 'octagon'
if (someCondition) iconClass = `${iconClass} some-other-class`

cdaringe commented on src/components/native/StopStartButton/StopStartButton.jsx:
```diff
@@ -4,11 +4,11 @@ import Icon from '../../suir/icon/Icon'
+    iconClass += ' ei arrow_triangle-right_alt2 stop-start-button'
   }
   return (
     <Icon className={` ${iconClass} ${props.className}`} style={props.style} onClick={props.onClick} />
```
if `props.className` is undefined, the class `undefined` will get rolled into the className.

`className={` ${iconClass} ${props.className}`}` should probably roll into

```js
if (props.className) iconClass = `${iconClass} ${props.className}`
<Icon className={iconClass} ... />
```

andrewleyva replied:
Hmm, yeah didn't focus too much on refactor, just namespacing but this who block can easily be cleaned up and benefit from templating. Gonna move all the classes into the default except the two conditional ones. Too much duplication here.

cdaringe commented on src/styles/components/tag-button.css:
```diff
@@ -8,163 +8,168 @@
-  cursor: pointer;
-}
-.tagContainer .tagArrow {
+.tagContainer {
```
should this be `.octagon.tagContainer` ?

andrewleyva replied:
yes. good catch.

cdaringe commented:
all of these commits need to be rebased/squashed into the semantic-release friendly format

ggascoigne commented on src/styles/components/micro-card.css:
```diff
@@ -20,7 +20,7 @@
   }
 
-  &.state__indicator {
+  & .state__indicator {
```
is this extra space fixing a bug or introducing one?

andrewleyva replied:
It's intentional. I guess you could say it's fixing a bug - meaning the css coding was odd.
Nesting revealed this potential bug. The css class microcard is used at multiple levels. 'state_indicator' initially was attached to the inner microcard class (not explicitly, but because it was the only existing css chain). By namespacing octagon at the same level as the outer microcard there was no no longer a valid css chain.
Before: .microcard.state__indicator (outer microcard)
After: .octagon.microcard.state__indicator (did not work, references inner microcard)
Fix: .octagon.microcard .state__indicator (references outer microcard.)

This is probably due for a refactor and I plan to take care of that in the stateless/view-layer upgrade story.

<img width="507" alt="screen shot 2018-02-20 at 12 27 13 pm" src="https://user-images.githubusercontent.com/1317206/36447750-2903ff8a-163a-11e8-9809-cc88b8e3bdd8.png">

ggascoigne commented on src/styles/components/pagination-control.css:
```diff
@@ -6,104 +6,108 @@
+  & .pagination__page-total {
+    margin-left: var(--paginationSpaceBetweenWords);
+  }
 }
```
css files should end with a newline.

andrewleyva replied:
👍

cdaringe replied:
cssnext is nice isnt it? :)

wa11-e commented:
Woohoo! new surge deployment available for viewing! :tada: [surgereviewiptatjkwxiakomk.surge.sh](http://surgereviewiptatjkwxiakomk.surge.sh)

wa11-e commented:
Woohoo! new surge deployment available for viewing! :tada: [surgereviewtfswgpqbpgkmxxv.surge.sh](http://surgereviewtfswgpqbpgkmxxv.surge.sh)

andrewleyva commented:
@cdaringe @ggascoigne Reviews Addressed:
- added newlines to css files that were lacking
- updated how css classes are added using templating
- added missing namespace to tag-buttoncss
- confirmed that micro-card css is correct

cdaringe commented:
hey andrew, the commits still need the re work.  i suppose id accept "squash and merge" from GitHub, but would prefer the commit(s) tidied in advanced

andrewleyva commented:
yeah i was planning on squashing and tidying commits because there are so
many commits.

On Tue, Feb 20, 2018 at 5:24 PM Christopher Dieringer <
notifications@github.com> wrote:

> hey andrew, the commits still need the re work. i suppose id accept
> "squash and merge" from GitHub, but would prefer the commit(s) tidied in
> advanced
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Tripwire/octagon/pull/168#issuecomment-367181441>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ABQZVvVKSAJ5JRaalwqiVuATKqdarnEAks5tW3AygaJpZM4SGIq5>
> .
>
-- 
Andrew Leyva
Front End Developer

(c) 240-672-7528
(e) andrew.leyva@gmail.com
(w) www.andrewleyva.com

<http://www.linkedin.com/in/andrewleyva>
<http://www.twitter.com/andrewleyva> <http://jsfiddle.net/user/andrewleyva/>
<http://stackoverflow.com/users/1636286/andrew-leyva>
